opacus/optimizers/optimizer.py

```diff
@@ -1,12 +1,37 @@
 from __future__ import annotations
 
-from typing import Callable, List, Optional
+from typing import Callable, List, Optional, Union
 
 import torch
 from torch import nn
 from torch.optim import Optimizer
 
 
+def _label_processed(obj: Union[torch.Tensor, List[torch.Tensor]]):
+    if isinstance(obj, torch.Tensor):
+        obj.processed = True
```

Comment:
nit: make this private attribute and more explicit. `_opacus_processed` ?

Reply:
+ on making private

On "opacus_processed" vs "processed" - do you think renaming make sense even if we only attach this to opacus-owned attributes (namely `p.grad_sample` and `p.summed_grad`)

Reply:
probably not for grad_sample since it is not necessarily for opacus alone. It is an attribute of module after all.

On the other hand, processed was solely for use by opacus? But then again, one could argue it is an attribute of the module irrespective of who uses it. In conclusion, it is a weak nit, and I am happy to withdraw it :)

Reply:
I vote for `_opacus_processed` and request short explanation in the docstring (why we need this label and why it is removed after zero_grad).

Reply:
Wait, we add this field to Tensor, not to tensor.grad_sample, right?

Reply:
Nope, we add it to `p.grad_sample` (or, in case of a multiple accumulated batches, where `.grad_sample` is a list - each tensor in that list)

it's `p.grad_sample._processed` and `p.summed_grad._processed`, never `p._processed`

Reply:
> it's p.grad_sample._processed and p.summed_grad._processed, never p._processed

Oh yes! I totally missed this. In that case, I completely withdraw my nit. For some reason, I was under the impression that it will be `p._processed`.  `grad_sample` is created by us, so `p.grad_sample._processed` is more than sufficient.